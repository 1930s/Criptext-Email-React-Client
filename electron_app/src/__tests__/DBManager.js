/* eslint-env node, jest */

const DBManager = require('../DBManager');
const fs = require('fs');

describe('Test DBManager', () => {

  it('should add email to db', async () => {
    await DBManager.insertRow('email', {
      key: 'hdnfgdgsd',
      threadId: 'hdnfgdgsd',
      s3Key: 'hdnfgdgsd',
      unread: false,
      secure: true,
      content: 'Hello there',
      preview: 'Hello there',
      subject: 'Greetings',
      delivered: 0,
      date: '2013-10-07 08:23:19.120',
      isTrash: false,
      isDraft: false,
    })
    
    const email = await DBManager.getRow('email', {
      threadId: 'hdnfgdgsd'
    })
    expect(email).toMatchSnapshot();
  });

  afterAll(() => {
    DBManager.closeDB();
    fs.unlink('./src/__tests__/test.db');
  });

});