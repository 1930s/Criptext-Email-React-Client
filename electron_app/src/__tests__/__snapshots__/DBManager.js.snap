// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DBManager should add email to db 1`] = `
Array [
  Object {
    "content": "Hello there",
    "date": "2013-10-07 08:23:19.120",
    "delivered": 0,
    "isDraft": 0,
    "isTrash": 0,
    "key": "hdnfgdgsd",
    "preview": "Hello there",
    "s3Key": "hdnfgdgsd",
    "secure": 1,
    "subject": "Greetings",
    "threadId": "hdnfgdgsd",
    "unread": 0,
  },
]
`;
